### PR TITLE
fix: user names update

### PR DESCRIPTION
**Pull request by SanketD92**
Issue #, if available:
GALI-1386

Description of changes:
Update first and last names to keep in sync with IdP.

Checklist:

- [x] Have you successfully deployed to an AWS account with your changes?
- [x] Have you written new tests for your core changes, as applicable?
- [x] Have you successfully tested with your changes locally?

AS review ticket id:

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: addons/addon-base-rest-api/packages/services/lib/authentication-providers/built-in-providers/cogito-user-pool/__tests__/provider-service.test.js [hunk 1]
```diff
@@ -125,6 +126,123 @@ describe('ProviderService', () => {
...
+        usernameInIdp: username,
+        email: username,
+        rev: 2,
+      };
+      const expected = {
+        firstName: 'Jen',
+        lastName: 'Doe',
+        rev: 2,
+        uid: 'sample-user-uid',
+      };
```
Comment (maghirardelli) [inline addons/addon-base-rest-api/packages/services/lib/authentication-providers/built-in-providers/cogito-user-pool/__tests__/provider-service.test.js:202]: Can you add a test case that matches the setup that led to this bug being found? The values would be something like this:
```
// from cognito
const userAttributes = {
        username,
        usernameInIdp: username,
        identityProviderName: 'Cognito Native Pool',
        isSamlAuthenticatedUser: false,
        isNativePoolUser: true,
        firstName: 'Jen',
        lastName: 'Doe',
        email: 'jen+doe@example.com',
      };
// From DDB
const existingUser = {
        identityProviderName: 'Cognito Native Pool',
        uid: 'sample-user-uid',
        username,
        firstName: 'jen+doe',
        lastName: 'jen+doe',
        isNativePoolUser: true,
        isSamlAuthenticatedUser: false,
        usernameInIdp: username,
        email: username,
        rev: 2,
};
const expected = {
        firstName: 'Jen',
        lastName: 'Doe',
        rev: 2,
        uid: 'sample-user-uid',
};
```

File: addons/addon-base-rest-api/packages/services/lib/authentication-providers/built-in-providers/cogito-user-pool/provider-service.js [hunk 2]
```diff
@@ -168,18 +167,33 @@ class ProviderService extends Service {
...
+
+      // When IdP users are created via SWB UI, by default we use the username part of provided email address as first and last names
+      // To update these default names, we extract the mapped attribute values coming from the Cognito token
+      // Some IdPs send email value in their firstName and lastName attribute fields, so splitting string to ignore domain
+      const updateIfDifferent = attribName => {
+        if (
+          !_.isUndefined(userAttributes[attribName]) &&
+          existingUser[attribName] !== userAttributes[attribName].split('@')[0]
+        ) {
+          updatedAttribs[attribName] = userAttributes[attribName].split('@')[0];
```
Comment (maghirardelli) [inline addons/addon-base-rest-api/packages/services/lib/authentication-providers/built-in-providers/cogito-user-pool/provider-service.js:179]: Why do we need the split on @? Do we expect Cognito or IdP values to have saved the names as emails? I did not observe that.

File: addons/addon-base-rest-api/packages/services/lib/authentication-providers/built-in-providers/cogito-user-pool/__tests__/provider-service.test.js (hunk 1 above)
Comment (SanketD92) [reply addons/addon-base-rest-api/packages/services/lib/authentication-providers/built-in-providers/cogito-user-pool/__tests__/provider-service.test.js:202]: This test case is exactly the same (basically we update the names if they're not equal to IdP-provided values).

File: addons/addon-base-rest-api/packages/services/lib/authentication-providers/built-in-providers/cogito-user-pool/provider-service.js (hunk 2 above)
Comment (SanketD92) [reply addons/addon-base-rest-api/packages/services/lib/authentication-providers/built-in-providers/cogito-user-pool/provider-service.js:179]: There are external IdPs (like Auth0 dev) that attaches email in the firstname value.

File: addons/addon-base-rest-api/packages/services/lib/authentication-providers/built-in-providers/cogito-user-pool/__tests__/provider-service.test.js (hunk 1 above)
Comment (SanketD92) [reply addons/addon-base-rest-api/packages/services/lib/authentication-providers/built-in-providers/cogito-user-pool/__tests__/provider-service.test.js:202]: This test case also covers the case where name value sent by IdP is in email format.

Comment (SanketD92) [reply addons/addon-base-rest-api/packages/services/lib/authentication-providers/built-in-providers/cogito-user-pool/__tests__/provider-service.test.js:202]: Added test case

**Review by maghirardelli — APPROVED** [review]
Looks good!

**Pull request merged**